docs/rules/no-runloop.md

```diff
@@ -0,0 +1,71 @@
+# ember/no-runloop
+
+<!-- end auto-generated rule header -->
+
+Ember's runloop functions are not lifecycle-aware and don't ensure that an object's async is cleaned up. It is recommended to use [`ember-lifeline`](https://ember-lifeline.github.io/ember-lifeline/), [`ember-concurrency`](http://ember-concurrency.com/docs/introduction/), or [`@ember/destroyable`](https://rfcs.emberjs.com/id/0580-destroyables/) instead.
+
+## Rule Details
+
+This rule disallows usage of `@ember/runloop` functions.
+
+## Examples
+
+Example of **incorrect** code for this rule:
+
+```ts
+import Component from '@glimmer/component';
+import { run } from '@ember/runloop';
+
+export default class MyComponent extends Component {
+  constructor() {
+    super(...arguments);
+
+    run.later(() => {
+      this.set('date', new Date());
+    }, 500);
+  }
+}
+```
+
+Example of **correct** code for this rule using `ember-lifeline`:
+
+```js
+import Component from '@glimmer/component';
+import { runTask, runDisposables } from 'ember-lifeline';
+
+export default class MyComponent extends Component {
+  constructor(...args) {
+    super(...args);
+
+    runTask(
+      this,
+      () => {
+        this.set('date', new Date());
+      },
+      500
+    );
+  }
+
+  willDestroy(...args) {
+    super.willDestroy(...args);
+
+    runDisposables(this);
+  }
+}
+```
+
+## Configuration
+
+If you have `@ember/runloop` functions that you wish to allow, you can configure this rule to allow specific methods. The configuration takes an array of strings, where the strings must be names of runloop functions.
+
+```ts
+module.exports = {
+  rules: {
+    'ember/no-runloop': ['error', ['debounce', 'begin', 'end']],
```

Comment:
I'm thinking we should probably change this schema to an object with an array property called `allowlist`. That makes it more clear what this list is, and also make it easier for us to add more options in the future.